packages/amplify-cli/src/lib/init-steps/preInitSetup.js

```diff
@@ -0,0 +1,121 @@
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+async function run(context) {
+  if (context.parameters.options.app) {
+    // Setting up a sample app
+    context.print.warning('Note: Amplify does not have knowledge or necessarily approve of url provided');
+    const repoUrl = context.parameters.options.app;
+    await validateGithubRepo(repoUrl);
+    await cloneRepo(repoUrl);
+    const packageManager = await getPackageManager();
+    await installPackage(packageManager);
+    await setLocalEnvDefaults(context);
+  }
+  return context;
+}
+
+/**
+ * Checks whether a url is a valid address
+ *
+ * @param str the url to validated
+ * @return true if str is a valid address, false otherwise
+ */
+async function isURL(str) {
+  const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|' + // domain name
+    '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+    '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+    '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
+  return pattern.test(str);
+}
+
+/**
+ * Checks whether a url is a valid remote github repository
+ *
+ * @param repoUrl the url to validated
+ * @throws error if url is not a valid remote github url
+ */
+async function validateGithubRepo(repoUrl) {
+  if (await isURL(repoUrl)) {
+    try {
+      execSync(`git ls-remote ${repoUrl}`, { stdio: 'ignore' });
+    } catch (e) {
+      throw Error('Invalid remote github url');
+    }
+  } else {
+    throw Error('Invalid url');
+  }
+}
+
+/**
+ * Clones repo from url to current directory (must be empty)
+ *
+ * @param repoUrl the url to be cloned
+ */
+async function cloneRepo(repoUrl) {
+  execSync(`git clone ${repoUrl} .`, { stdio: 'inherit' });
```

Comment:
I got this error: 

`fatal: destination path '.' already exists and is not an empty directory.`

You need to remove the `.`

Reply:
And after clone the repo, you need to call process.chdir() to change the process.cwd to the repo's root directory, otherwise you will get a bunch of errors.

Reply:
tested in an empty folder, and it's working end to end. 
I think this is the PM's decision if to allow the user to just run the command anywhere, or the user has to create an empty folder first and step into it and then run the command.

Reply:
Removed the stack trace